examples/examples.md

```diff
@@ -891,10 +891,9 @@ states:
 #### Description
 
 In this example we show off the states error handling capability. The workflow data input that's passed in contains
-missing order information that causes the function in the "ProvisionOrder" state to throw a runtime exception. With the "onError" expression we
-can transition the workflow to different error handling states depending on the error thrown. Each type of error
-in this example is handled by simple delay states, each including an error data filter which sets the exception info as their
-data output. If no error is caught the workflow can transition to the "ApplyOrder" state.
+missing order information that causes the function in the "ProvisionOrder" state to throw a runtime exception. With the "onErrors" definition we
+can transition the workflow to different error handling states. Each type of error
+in this example is handled by simple delay states. If no errors are encountered the workflow can transition to the "ApplyOrder" state.
 
```

Comment:
Since our default actions are OpenAPI-compatible, should we make a statement how the errors used in this example "Missing order item", "Missing order quantity" would relate to a Swagger specification of the action?

Reply:
they do not relate to swagger. even tho swagger defines how to invoke a service, the underlying runtime impl that actually executes it from the swagger def can be in many different languages and as such the errors that it can produce could vary . runtimes must provide the capabilities for this mapping and not our spec or swagger

Reply:
Agree that runtimes can cause additional errors that are not covered by the action definition. But how portable is a workflow spec when its error handling is runtime-specific?

Reply:
any runtime impl which choses to use SW lang for workflows will need to allow users to create a mapping btw. how errors are expressed in the workflow lang (domain-specific) to the internal errors. 

Most of the time this will be done by the runtimes catching an exception and wrapping it into a custom exception which will include a certain "code" that represents that type of exception. Workflow implementors should have this list of codes available and can use them.

The non-portability factor that I think you are seeing is that those codes might not be portable between runtimes, and there you are correct. When moving from one runtime to another two things could happen:
1) The new runtime provides the mapping to the codes expressed in the workflows 
2) Workflow implementors might have to adjust the codes in their workflows (not the best solution, but ok)

Either way the domain-specific error name stays the same. So yeah I think you do have a point so lets in further updates think maybe of better ways to handle this is there are ideas.